ctype.c:14: error: redefinition of typedef 'RParseCType' - previous declaration of 'RParseCType' was in libr/include/r_parse.h:80

### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | FreeBSD
| Architecture/bits of the file (mandatory)            | x86/32
| Error log            |  https://github.com/unixfreaxjp/dev/blob/master/FreeBSD-2019-057.md
| Problem affecting r2 versions     | 3.7.1, 3.8.0, 3.9.0, and recent
| Problem description     | ctype.c:14: error: redefinition of typedef 'RParseCType' - previous declaration of 'RParseCType' was in libr/include/r_parse.h:80 ; The typedef for RParseCType was coded in https://github.com/radareorg/radare2/commit/24d3a9cb5d5497c6bd59df9a22792339f873b594#diff-65addf422e419aaa267fbe7de38e57aa and being typedef’Ed again in https://github.com/radareorg/radare2/commit/24d3a9cb5d5497c6bd59df9a22792339f873b594#diff-d070ab2cfdd810d61567be662fa72e87 ; this created the compile error; the coder was:  (deleted)
| Workariund                             | don’t use ctype.c in libr/parser, comment it in meson script too.

### Expected behavior

Compiled successfully

### Actual behavior

Compilation/build failed with Error.

### Steps to reproduce the behavior 

- no shell C variable or macro set, just default setting
 - run sys/install.sh as usual and the problem described in the log above always happened

### Solution proposed.

Please avoid double typedef 'RParseCType'

cc: @radare @xvilka @devnexen @unixfreaxjp

--- a/libr/parse/ctype.c
+++ b/libr/parse/ctype.c
@@ -4,14 +4,14 @@
 
 #include <mpc.h>
 
-typedef struct r_parse_ctype_t {
+struct r_parse_ctype_t {
 	mpc_parser_t *integerlit;
 	mpc_parser_t *identifier;
 	mpc_parser_t *qualifier;
 	mpc_parser_t *pointer;
 	mpc_parser_t *array;
 	mpc_parser_t *type;
-} RParseCType;
+};
 
 #define ALL_PARSERS(ctype) ctype->integerlit, ctype->identifier, ctype->qualifier, ctype->pointer, ctype->array, ctype->type
 #define ALL_PARSERS_COUNT 6

